### DISCUSSION TITLE
Ingestion Threashold per API-Key

### DISCUSSION THREAD

**Discussion by Sebbl22** [datalust/seq-tickets#1197] [Ideas]
We had some issues in the past with Seq with no remaining disk space on the log drive, because the amount of ingested log messages exeeded our predictions.
This usually happend, because one application had an issue and ingested a lot of error messages, e.g. 10 Million Messages per day instead of just a couple of 1000s.

To prevent this, it would be nice to be able to configure threasholds on a per-API-Key basis. I am not sure, how the available options could look like, but I imagine to be able to configure threasholds based on # of messages ingested in 24h and the size of those messages.

The goal would be to prevent a singe application to bring down our Seq server.

Maybe this would be worth considering or there are other options available.

**Comment by nblumhardt** [comment]
Thanks for the suggestion, Sebastian. We've made some progress in this area via the work underpinning the _Ingestion_ screen (we now track ingestion metrics per-API-key), so this sounds like a natural extension of that. We'll give this some thought 👍
